process/economics/economicsData.go

```diff
@@ -289,6 +301,20 @@ func (ed *economicsData) ComputeTxFee(tx process.TransactionWithFeeHandler) *big
 	return ed.ComputeMoveBalanceFee(tx)
 }
 
+// SplitTxGasInCategories returns the gas split per categories
+func (ed *economicsData) SplitTxGasInCategories(tx process.TransactionWithFeeHandler) (gasLimitMove, gasLimitProcess uint64) {
+	gasLimitMove = ed.ComputeGasLimit(tx)
+	gasLimitProcess = tx.GetGasLimit() - gasLimitMove
+	if gasLimitProcess < 0 {
```

Comment:
gasLimitProcess cannot be < 0 as it is uint64. Use core.SafeSubUint64 instead

Reply:
good catch